docs/_static/js/banner.js

```diff
@@ -1,39 +1,69 @@
-      <div>
-        <strong>Wagtail Space NL</strong><br> 15 – 17 June 2022 in Arnhem, The Netherlands
-      </div>
-    </a>
-    <button class="wagtailspace-close" aria-label="Hide banner" aria-expanded="true">×</span>
-  </div>
-  `);
+  /**
+   * Creates the promotion banner based on the element in the DOM.
+   *
+   * @returns {void}
+   */
+  function createBanner() {
+    const template = document.querySelector('[data-promotion-banner]');
+    if (!template) return;
+    const dateUntil = new Date(template.dataset.dateUntil || 'INACTIVE');
+    if (!(dateUntil instanceof Date) || Number.isNaN(dateUntil.valueOf())) {
+      return;
+    }
+    // Pause any promotional banner for X days when manually cleared
+    const COOKIE_DURATION = Math.max(
+      Number(template.dataset.clearDuration || 30),
+      30,
+    );
+    const COOKIE_NAME = 'WAGTAIL_PROMOTION_BANNER_CLOSED';
```

Comment:
The cookie name was unique per event to make sure new banners display regardless of the user closing the previous event banners.

Maybe we should add a comment to explain that?

Reply:
Good call. I can make this pull from a data attribute also so it's easier to update. 

Alternatively. I can append the expiry date to the cookie name. This way we get a unique Cooke name for any changes promotion expiry.

Reply:
Just updated this to have a more unique cookie name for each promotion based on the date the promotion is meant to run until.

@allcaps